File: solutions/c#/MusicStreamingService/Song.cs
namespace MusicStreamingService
{
    public class Song
    {
        public string Id { get; }
        public string Title { get; }
        public string Artist { get; }
        public string Album { get; }
        public int Duration { get; }

        public Song(string id, string title, string artist, string album, int duration)
        {
            Id = id;
            Title = title;
            Artist = artist;
            Album = album;
            Duration = duration;
        }
    }
}

File: solutions/java/src/musicstreamingservice/Song.java
package musicstreamingservice;

import java.time.Duration;
import java.util.UUID;

public class Song {
    private final String id;
    private final String title;
    private final Artist artist;
    private final Album album;
    private final Duration duration;

    public Song(String title, Artist artist, Album album, Duration duration) {
        this.id = UUID.randomUUID().toString();
        this.title = title;
        this.artist = artist;
        this.album = album;
        this.duration = duration;
    }

    public String getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public Artist getArtist() {
        return artist;
    }

    public String getArtistName() {
        return artist.getName();
    }

    public Album getAlbum() {
        return album;
    }
}
